src/module-elasticsuite-virtual-category/Model/Rule.php

```diff
@@ -139,15 +139,21 @@ public function __toString(): string
     public function getCategorySearchQuery($category, $excludedCategories = []): ?QueryInterface
     {
         $query         = null;
+        $rootCategory  = null;
 
         if (!is_object($category)) {
             $category = $this->categoryFactory->create()->setStoreId($this->getStoreId())->load($category);
         }
 
         if (!in_array($category->getId(), $excludedCategories)) {
             $excludedCategories[] = $category->getId();
-
-            if ((bool) $category->getIsVirtualCategory() && $category->getIsActive()) {
+            if ($category->getIsVirtualCategory() && $category->getVirtualCategoryRoot() !== null
+                && !empty($category->getVirtualCategoryRoot())) {
+                $rootCategoryId = $category->getVirtualCategoryRoot();
+                $rootCategory = $this->categoryFactory->create()->setStoreId($this->getStoreId())->load($rootCategoryId);
+                $catId = $rootCategory->getId();
+            }
+            if ((bool) $category->getIsVirtualCategory() && $category->getIsActive() && isset($catId)) {
```

Comment:
Not a fan of the approach since it integrates part of the logic of checking the existence of the root category in \Smile\ElasticsuiteVirtualCategory\Model\Rule::getVirtualRootCategory.
I would change the logic of \Smile\ElasticsuiteVirtualCategory\Model\Rule::getVirtualCategoryQuery so that it returns a null query if \Smile\ElasticsuiteVirtualCategory\Model\Rule::getVirtualRootCategory returns null and change this section of code in \Smile\ElasticsuiteVirtualCategory\Model\Rule::getCategorySearchQuery 
with something like this : 
```
            if ($category->getIsActive()) {
                if ((bool) $category->getIsVirtualCategory()) {
                  $query = $this->getVirtualCategoryQuery($category, $excludedCategories, $category->getData('virtual_category_root'));
                }
                if (!$query) {
                  if ($category->getId() {
                    $query = $this->getStandardCategoryQuery($category, $excludedCategories);
                  }
                }
            }
            if ($query && $category->hasChildren()) {
                $query = $this->addChildrenQueries($query, $category, $excludedCategories);
            }
```
But it's not perfect, and still a bit more complex.

On an additional note, your fix will apply when visiting the virtual category, but won't have any effect on parent categories, so we might want to address the problem as a whole by looking at what would happen in \Smile\ElasticsuiteVirtualCategory\Model\Rule::addChildrenQueries

Reply:
Conclusion: we need to discuss this.